lockfile/src/parse_depfile.rs

```diff
@@ -83,11 +86,16 @@ pub fn parse_depfile(
         }
     }
 
-    // Generate lockfile if path might be a manifest and feature and option are
-    // enabled.
+    // Attempt to generate a lockfile for likely manifests when feature and option
+    // are enabled. This is a best effort attempt for files that are known at this
+    // point to not be a valid/parseable lockfile but may parse as a manifest with
+    // a non-standard name.
     #[cfg(feature = "generator")]
-    if let Some(generation_path) = _generation_path.filter(|_| maybe_manifest) {
-        return Ok(generate_lockfile(&generation_path, &path, format, parser)?);
+    if let Some(generation_path) = _generation_path.filter(|_| !maybe_lockfile || maybe_manifest) {
+        let result = generate_lockfile(&generation_path, &path, format, parser);
+        if result.is_ok() {
+            return Ok(result?);
+        }
```

Comment:
1. This would be better written as a `match` (since it avoids double-checking `result` with both `.is_ok()` and `?`).
2. Sometimes we want to return the error (specifically when `maybe_manifest` is true)
3. As mentioned [here](https://github.com/phylum-dev/cli/pull/1337#discussion_r1450951131), some parsers don't support generation, so we might as well skip it.

This patch fixes all 3 issues.

```suggestion
        if parser.generator().is_some() {
            match generate_lockfile(&generation_path, &path, format, parser) {
                Ok(depfile) => return Ok(depfile),
                // Discard errors for unknown files. The error from the lockfile parser can be used
                // instead.
                Err(_) if !maybe_manifest => {},
                Err(err) => return Err(err.into()),
            }
        }
```

Reply:
This suggestion was accepted.